src/index.ts

```diff
@@ -26,8 +25,8 @@ export function PluginPure(options: PureAnnotationsOptions): Plugin {
           return
         }
 
-        // quick check if any of the functions are in the code
-        if (!options.functions.some(func => code.includes(func))) {
+        // quick check for early return
+        if (options.functions.every(func => typeof func === 'string' && !code.includes(func))) {
```

Comment:
I think we could test the code with the regexp for early return.

Reply:
It is unsafe to check regexp here. For example, `/$definePage$/` can match a function named `definePage`, but testing against the entire code text `function definePage()...` results `false`.